packages/progress-monitor/src/browser/progress-dialog.ts

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from "inversify";
+import { Message } from '@phosphor/messaging';
+import { Disposable } from "@theia/core";
+import { Widget, BaseWidget, Key } from "@theia/core/lib/browser";
+import { ProgressWidget } from "./progress-widget";
+
+@injectable()
+export class ProgressDialog extends BaseWidget {
+
+    protected readonly contentNode: HTMLDivElement;
+
+    protected isOpen: boolean = false;
+
+    constructor(
+        @inject(ProgressWidget) readonly widget: ProgressWidget
+    ) {
+        super();
+
+        this.contentNode = document.createElement("div");
+        this.contentNode.classList.add('theia-ProgressContent');
+
+        this.node.classList.add('theia-ProgressWidgetDialog');
+
+        this.node.appendChild(this.contentNode);
+
+        this.toDispose.push(widget);
+
+        this.update();
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        super.onUpdateRequest(msg);
+        this.widget.update();
+    }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        Widget.attach(this.widget, this.contentNode);
+        this.toDisposeOnDetach.push(Disposable.create(() =>
+            Widget.detach(this.widget)
+        ));
+
+        this.addKeyListener(document.body, Key.ESCAPE, e => {
+            this.close();
+            this.isOpen = false;
+        });
+    }
+
+    protected onActivateRequest(msg: Message): void {
+        super.onActivateRequest(msg);
+        this.widget.activate();
+    }
+
+    async toggleOpen(): Promise<void> {
+        if (this.isOpen) {
+            this.close();
+        } else {
+            const thisStatusBarItem = document.getElementsByClassName("theia-ProgressStatusBar").item(0) as HTMLElement;
+            Widget.attach(this, thisStatusBarItem);
+            this.activate();
+        }
+        this.isOpen = !this.isOpen;
```

Comment:
[minor]: This should be `this.isOpen = true` if you have used `this.isOpen = false` for the key listener.